overlord/snapstate/backend/setup.go

```diff
@@ -150,13 +146,13 @@ func (b Backend) SetupKernelSnap(instanceName string, rev snap.Revision, meter p
 		if err == nil {
 			return
 		}
-		if e := sysd.RemoveMountUnitFile(filepath.Join(dirs.GlobalRootDir, earlyMountDir)); e != nil {
+		if e := sysd.RemoveMountUnitFile(earlyMountDir); e != nil {
 			meter.Notify(fmt.Sprintf("while trying to clean up due to previous failure: %v", e))
 		}
 	}()
 
 	// Build kernel tree that will be mounted from initramfs
-	return kernel.EnsureKernelDriversTree(instanceName, rev, filepath.Join(dirs.GlobalRootDir, earlyMountDir))
+	return kernel.EnsureKernelDriversTree(instanceName, rev, earlyMountDir)
```

Comment:
only relevant for tests, but it may be useful to add dirs.StripRoot() to some of the symlink targets in `createModulesSubtree`, and `createFirmwareSubtree` which IIRC create absolute symlinks.